nf_core/pipeline-template/{{cookiecutter.name_noslash}}/.github/CONTRIBUTING.md

```diff
@@ -52,6 +53,76 @@ These tests are run both with the latest available version of `Nextflow` and als
 * Fix the bug, and bump version (X.Y.Z+1).
 * A PR should be made on `master` from patch to directly this particular bug.
 
-## Getting help
+## Getting Help
 
 For further information/help, please consult the [{{ cookiecutter.name }} documentation](https://nf-co.re/{{ cookiecutter.short_name }}/usage) and don't hesitate to get in touch on the nf-core Slack [#{{ cookiecutter.short_name }}](https://nfcore.slack.com/channels/{{ cookiecutter.short_name }}) channel ([join our Slack here](https://nf-co.re/join/slack)).
+
+## Pipeline Contribution Conventions
+
+To make the {{ cookiecutter.short_name }} code and processing logic more understandable for new contributors and to ensure quality, we semi-standardise the way the code and other contributions are written.
+
+### Adding a New Module
+
+If you wish to contribute a new module, please use the following coding standards:
+
+1. Define the corresponding input channel into your new process from the expected previous process channel
+2. Write the process block (see below).
+3. Define the output channel if needed (see below).
+4. Add any new flags/options to `nextflow.config` with a default (see below).
+5. Add any new flags/options to `nextflow_schema.json` with help text (with `nf-core schema build .`)
+6. Add any new flags/options to the help message (for integer/text parameters, print to help the corresponding `nextflow.config` parameter).
+7. Add sanity checks for all relevant parameters.
+8. Add any new software to the `scrape_software_versions.py` script in `bin/` and the version command to the `scrape_software_versions` process in `main.nf`.
+9. Do local tests that the new code works properly and as expected.
+10. Add a new test command in `.github/workflow/ci.yaml`.
+11. If applicable add a [MultiQC](https://https://multiqc.info/) module.
+12. Update MultiQC config `assets/multiqc_config.yaml` so relevant suffixes, name clean up, General Statistics Table column order, and module figures are in the right order.
+13. Optional: Add any descriptions of MultiQC report sections and output files to `docs/output.md`.
+
+More details are as follow
+
+### Default Values
+
+Default values should go in `nextflow.config` under the `params` scope, and `nextflow_schema.json` (latter with `nf-core schema build .`)
+
+### Default Processes Resource Requirements
+
+Defining recommended 'minimum' resource requirements (CPUs/Memory) for a process should be defined in `conf/base.config`. This can be utilised within the process using `${task.cpu}` or `${task.memory}` variables in the `script:` block.
```

Comment:
```suggestion
Defining sensible defaults for process resource requirements (CPUs / memory / time) for a process should be defined in `conf/base.config`. This can be utilised within the process using `${task.cpu}` or `${task.memory}` variables in the `script:` block.
```

Reply:
Would also be good to add mention of tagging processes with labels and trying to stick to the standard nf-core label names where possible (see eg. [the rnaseq pipeline](https://github.com/nf-core/rnaseq/blob/bc5fc76f40b2da6082a854927184c9d6e5060393/conf/base.config#L23-L40)).

Reply:
Hmm, I actually wanted to ask about the latter - I saw that rnaseq is apparently the 'standard', however it doesn't account for single CPU processes... is this something I should bring up on slack for wider discussion?

In eager I have basically what is in rnaseq, but with two versions: `sc` for single core and `mc` for multicore.

Reply:
The base config (ie no label) can be used for single CPU processes: https://github.com/nf-core/rnaseq/blob/bc5fc76f40b2da6082a854927184c9d6e5060393/conf/base.config#L14-L16

Reply:
Hmm OK. And I guess you do process specific increases for when they need more memory?